core/templates/pages/contributor-dashboard-page/services/contribution-and-review.service.ts

```diff
@@ -170,6 +182,143 @@ export class ContributionAndReviewService {
     );
   }
 
+  async fetchTranslationSuggestionsAsync(
+      explorationId: string
+  ): Promise<FetchSuggestionsResponse> {
+    const explorationBackendResponse = await this.
+      readOnlyExplorationBackendApiService.fetchExplorationAsync(
+        explorationId, null);
+    return(
+      this.contributionAndReviewBackendApiService.
+        fetchSuggestionsAsync(
+          'REVIEWABLE_TRANSLATION_SUGGESTIONS',
+          null,
+          0,
+          AppConstants.SUGGESTIONS_SORT_KEY_DATE,
+          explorationId
+        ).then((fetchSuggestionsResponse) => {
+          const explorationBackendDict = this.explorationObjectFactory.
+          createBackendDictFromExplorationBackendResponse(
+            explorationBackendResponse);
+
+          const exploration: Exploration = this.explorationObjectFactory.
+            createFromBackendDict(explorationBackendDict);
+          const sortedTranslationCards = this.sortTranslationSuggestionsByState(
+            fetchSuggestionsResponse.suggestions,
+            exploration.getStates(),
+            explorationBackendDict.init_state_name
+          );
+          // eslint-disable-next-line
+          const responseSuggestionIdToDetails: {[key: string]: any} = {};
+          const targetIdToDetails = fetchSuggestionsResponse.target_id_to_opportunity_dict;
+          sortedTranslationCards.forEach((suggestion) => {
+          const suggestionDetails = {
+            suggestion: suggestion,
+            details: targetIdToDetails[suggestion.target_id]
+          };
+          responseSuggestionIdToDetails[suggestion.suggestion_id] = suggestionDetails;
+          });
+          return {
+            suggestionIdToDetails: responseSuggestionIdToDetails,
+            more: false
+          };
+        })
+    );
+  }
+
+  // Function to sort translation cards by state.
+  sortTranslationSuggestionsByState(
+      // eslint-disable-next-line
+      translationSuggestions: any[],
+      states: States,
+      initStateName: string | null
+      // eslint-disable-next-line
+  ): any[] {
```

Comment:
Why is the type `any`? Isn't this `SuggestionBackendDict`?

Reply:
I initially tried with `SuggestionBackendDict`, but the ts tests were failing `https://github.com/oppia/oppia/actions/runs/5456888895/job/14773325903#step:9:134`